src/Compilers/CSharp/Test/Emit/CodeGen/SwitchTests.cs

```diff
@@ -1947,6 +1947,160 @@ .maxstack  2
 ");
         }
 
+        [Fact]
+        public void SwitchOnNullableInt64WithInt32Label()
+        {
+            var text = @"public static class C
+{
+    public static bool F(long? x)
+    {
+        switch (x)
+        {
+            case 1:
+                return true;
+            default:
+                return false;
+        }
+    }
+    static void Main()
+    {
+        System.Console.WriteLine(F(1));
+    }
+}";
+            var compVerifier = CompileAndVerify(text, expectedOutput: "True");
+            compVerifier.VerifyIL("C.F(long?)",
+@"
+{
+        // Code size       28 (0x1c)
+        .maxstack  2
+        .locals init (long? V_0,
+                      long V_1)
+        IL_0000:  ldarg.0
+        IL_0001:  stloc.0
+        IL_0002:  ldloca.s   V_0
+        IL_0004:  call       ""bool long?.HasValue.get""
+        IL_0009:  brfalse.s  IL_001a
+        IL_000b:  ldloca.s   V_0
+        IL_000d:  call       ""long long?.GetValueOrDefault()""
+        IL_0012:  stloc.1
+        IL_0013:  ldloc.1
+        IL_0014:  ldc.i4.1
+        IL_0015:  conv.i8
+        IL_0016:  bne.un.s   IL_001a
+        IL_0018:  ldc.i4.1
+        IL_0019:  ret
+        IL_001a:  ldc.i4.0
+        IL_001b:  ret
+}"
+            );
+        }
+
+        [Fact]
+        public void SwitchOnNullableWithNonConstant()
+        {
+            var text = @"public static class C
+{
+    public static bool F(int? x)
+    {
+        int i = 4;
+        switch (x)
+        {
+            case i:
+                return true;
+            default:
+                return false;
+        }
+    }
+    static void Main()
+    {
+        System.Console.WriteLine(F(1));
+    }
+}";
+            var compilation = base.CreateCSharpCompilation(text);
+
+            var expected = Diagnostic(ErrorCode.ERR_ConstantExpected, "case i:");
```

Comment:
Please use baseline style as in CS0469_NoImplicitConversionWarning method for example, i.e.
```
            // (10,17): warning CS0469: The 'goto case' value is not implicitly convertible to type 'System.DayOfWeek'
            //                 goto case 1; // warning CS0469: The 'goto case' value is not implicitly convertible to type 'System.DayOfWeek'
                Diagnostic(ErrorCode.WRN_GotoCaseShouldConvert, "goto case 1;").WithArguments("System.DayOfWeek"))
```

If you run test without passing any arguments to VerifyDiagnostics, the test will fail and it will print reported diagnostics in the form that you can simply copy and paste as the baseline.